lib/pages/contacts/data/datasource/contact_data_source.dart

```diff
@@ -0,0 +1,14 @@
+
+import 'package:fluffychat/entity/contact/contact.dart';
+
+abstract class ContactDataSource {
+  Future<List<Contact>> getContacts({withThumbnail = false});
+}
+
+enum LocalDataSourceType {
+  device,
+}
+
+enum NetworkDataSourceType {
+  tomclient,
+}
```

Comment:
why we need these `enum`?

Reply:
for mapping datasources in repository layer, i do this, so that the repository know which datasource it use